src/ComputeSharp.D2D1/Shaders/Interop/Helpers/Rectangle64.cs

```diff
@@ -112,15 +112,15 @@ public void Union(Rectangle64 rectangle)
     /// <param name="rectangle">The input <see cref="Rectangle64"/> instance to compute the intersection with.</param>
     public void Intersect(Rectangle64 rectangle)
     {
-        long maxLeft = Math.Max(this.x, rectangle.x);
-        long maxTop = Math.Max(this.y, rectangle.y);
-        long width = Math.Max(Math.Min(this.Right, rectangle.Right) - maxLeft, 0);
-        long height = Math.Max(Math.Min(this.Bottom, rectangle.Bottom) - maxTop, 0);
+        long left = Math.Max(this.x, rectangle.x);
+        long top = Math.Max(this.y, rectangle.y);
+        long right = Math.Max(Math.Min(this.Right, rectangle.Right), 0);
```

Comment:
why are you maxing against 0? rectangles can be in negative coordinate space, D2D does that all the time

Reply:
Whoops, fixed, thanks! 😄